### PR TITLE
Use webworkers for expensive tasks & faster CSV parsing.

### PR DESCRIPTION
**Pull request by maartenvn**
## Webworkers

Some expensive calculations can freeze the main thread (eg: parsing fragments/semantics). To prevent this a custom web worker was written.

In order to make the usage of a webworker as convenient as possible, [`comlink-loader`](https://github.com/GoogleChromeLabs/comlink-loader) was used to automatically transpile `*.worker.ts` files into webworkers. You can call the functions in this webworker as regular TypeScript function inside other files.

**Note for developers: all functions in a worker must be async and have the Promise return type**

## CSV Parser

The D3 CSV parser is not the most efficient CSV parser and causes the application to freeze when loading data (due to parsing on the main thread). This PR replaces the D3 CSV parser by [PapaParse](https://www.papaparse.com/), which is much faster. The webworker feature is used to parse in a webworker, preventing freezing the main thread.

## Better loading screen

The loading screen now shows the different loading tasks which can be handy for both the user & developer.

image

### CODE REVIEW

**Comment by maartenvn** [comment]
@rien #823 must be reviewed/merged before this PR.

File: web/src/composables/useWorker.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  };
+}
+
+/**
+ * Helper for creating a worker with a generic object of functions.
+ * @param fns Object of functions to be exposed to the worker.
+ */
+export function makeWorker(fns?: WorkerFns): void {
+  self.addEventListener("message", ({ data: { key, args } }) => {
+    const result = fns?.[key](...args);
```
[the remark this answers is not in the document]
Comment (maartenvn) [reply web/src/composables/useWorker.ts]: Resolved

File: web/package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
   "devDependencies": {
     "@types/d3": "7.4.0",
     "@types/luxon": "3.0.0",
+    "@types/papaparse": "^5.3.3",
```
Comment (rien) [inline web/package.json:15]:
```suggestion
    "@types/papaparse": "5.3.3",
```


Since enabling Renovate, we are pinning `devDependencies`.

**Review by rien — APPROVED** [review]
Looking good :ok_hand: 

Can you disable the `Unexpected any` warnings on the lines in the store where they are unavoidable?

Do you have an idea how much faster loading is now (ballpark)?

**Pull request merged**